server.go

```diff
@@ -884,6 +884,30 @@ func newServer(cfg *Config, listenAddrs []net.Addr, chanDB *channeldb.DB,
 		OutgoingBroadcastDelta: lncfg.DefaultOutgoingBroadcastDelta,
 		NewSweepAddr:           newSweepPkScriptGen(cc.wallet),
 		PublishTx:              cc.wallet.PublishTransaction,
+		ListTxnsAtHeight: func(height uint32) ([]*wire.MsgTx, error) {
+			blockHeight := int32(height)
+			transactions, err := cc.wallet.ListTransactionDetails(
```

Comment:
Why not just query for the transaction directly from the `wallet`? This may require us to surface a new `wtxmgr` method.

Reply:
I tried looking up the input with `FetchInputInfo` but that doesn't work because `btcWallet` doesn't think the funding outpoint is our input. Had a look at `wtxmgr/query` and I don't see much we could use for tx lookup?

Reply:
actually nvm we don't need this, use htlc as in your following comments 👍